server/rest/src/main/java/org/infinispan/rest/resources/ProtobufResource.java

```diff
@@ -0,0 +1,178 @@
+import org.infinispan.query.remote.client.ProtobufMetadataManagerConstants;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.cachemanager.RestCacheManager;
+import org.infinispan.rest.framework.ContentSource;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.rest.operations.exceptions.NoDataFoundException;
+import org.infinispan.rest.operations.exceptions.NoKeyException;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON;
+import static org.infinispan.rest.framework.Method.DELETE;
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.POST;
+import static org.infinispan.rest.framework.Method.PUT;
+
+/**
+ * Protobuf schema manipulation Resource
+ *
+ * @author Katia Aresti
+ * @since 11
+ */
+public class ProtobufResource extends BaseCacheResource implements ResourceHandler {
+
+   public ProtobufResource(InvocationHelper invocationHelper) {
+      super(invocationHelper);
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            // Key related operations
+            .invocation().methods(GET).path("/v2/protobuf").handleWith(this::getSchemasNames)
+            .invocation().methods(PUT, POST).path("/v2/protobuf/{schemaName}").handleWith(this::addOrEditSchema)
+            .invocation().methods(GET).path("/v2/protobuf/{schemaName}").handleWith(this::getSchema)
+            .invocation().method(DELETE).path("/v2/protobuf/{schemaName}").handleWith(this::deleteSchema)
+            .create();
+   }
+
+   private CompletionStage<RestResponse> getSchemasNames(RestRequest request) {
+      AdvancedCache<Object, Object> cache = invocationHelper.getRestCacheManager()
+            .getCache(ProtobufMetadataManager.PROTOBUF_METADATA_CACHE_NAME, request);
+
+      List<ProtoSchema> protoSchemas = cache.entrySet().stream()
+            .filter(cacheEntry -> !((String)cacheEntry.getKey()).endsWith(ProtobufMetadataManager.ERRORS_KEY_SUFFIX))
+            .map(cacheEntry -> {
+               String error = (String) cache
+                     .get(cacheEntry.getKey() + ProtobufMetadataManagerConstants.ERRORS_KEY_SUFFIX);
+               ProtoSchema protoSchema = new ProtoSchema();
+               protoSchema.name = cacheEntry.getKey().toString();
+               protoSchema.validation = error == null ? "success" : error;
+               return protoSchema;
+            })
+            .collect(Collectors.toList());
```

Comment:
done

Reply:
This is not fixed yet. It needs to be done using the blocking executor, because this cache is persisted. To be honest (not for this PR) the server/rest module needs blockhound added to find these automatically.